src/Agent.Plugins/RepositoryPlugin.cs

```diff
@@ -185,7 +185,31 @@ public override async Task RunAsync(AgentTaskPluginExecutionContext executionCon
             }
 
             ISourceProvider sourceProvider = SourceProviderFactory.GetSourceProvider(repo.Type);
-            await sourceProvider.GetSourceAsync(executionContext, repo, token);
+
+            int retryLimit = 5;
+            int currentRetry = 1;
+            TimeSpan delay = TimeSpan.FromSeconds(1);
+            while (currentRetry <= retryLimit)
+            {
+                try
+                {
+                    executionContext.Output($"Started #{currentRetry} attempt of GetSourceAsync.");
+                    await sourceProvider.GetSourceAsync(executionContext, repo, token);
+                    return;
+                }
+                catch (Exception ex)
+                {
+                    executionContext.Debug($"Failed #{currentRetry} attempt of GetSourceAsync.");
+                    executionContext.Debug(ex.ToString());
+                    currentRetry++;
+
+                    if (currentRetry > retryLimit)
+                    {
+                        throw;
+                    }
+                }
+                await Task.Delay(delay);
```

Comment:
I think this approach will add retries to everything, not just TFVC operations. I think a better approach would be to handle this in the TFVC CLI manager. I already did this for gets, but if we need it in more operations it should be pretty easy to build off of my pr here - https://github.com/microsoft/azure-pipelines-agent/pull/2972

Reply:
Adding on this, were you able to repro the issue that users were facing?  

I'm concerned that error was due to the naming of the workspace folder and if so a retry wouldn't work.

Let me know what you think.

Reply:
@shkiltos let's discuss offline, feel free to ping me

Reply:
@shkiltos to follow up, I don't think this is the right approach, I think we'd be better served handling this directly in the TFVC CLI Manager, rather than retrying everything